docs/examples/example_survival.ipynb

```diff
@@ -0,0 +1,259 @@
+    "\n",
+    "* **Censoring Support:** For handling this issue, we keep ``y`` as is. If the event was observed, the value represents when it occurred, and if not observed, it is the censoring time. We then augment the ``X`` dataset with an extra column (which won't be passed to the model), labeled ``\"censored\"``, indicating whether the observation was censored or not. This column aids in constructing the upper and lower bounds for the output label from the model.\n",
+    "\n",
+    "\n",
+    "We suggest leveraging the base classes from ``sklearn`` to ensure that some essential methods needed for the MetaLearner are pre-defined and can be utilized effectively."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import xgboost as xgb\n",
+    "from sklearn.base import BaseEstimator, RegressorMixin, check_is_fitted\n",
+    "import numpy as np\n",
+    "\n",
+    "\n",
+    "class SurvivalModel(RegressorMixin, BaseEstimator):\n",
+    "    def __init__(self, n_estimators=100, booster_params=None) -> None:\n",
+    "        self.booster_params = booster_params\n",
+    "        self.n_estimators = n_estimators\n",
+    "\n",
+    "    def fit(self, X, y):\n",
+    "        common_params = {\n",
+    "            \"objective\": \"survival:aft\",\n",
+    "            \"eval_metric\": \"aft-nloglik\",\n",
+    "            \"aft_loss_distribution\": \"normal\",\n",
+    "            \"aft_loss_distribution_scale\": 1.20,\n",
+    "        }\n",
+    "\n",
+    "        # X contains \"censored\" column\n",
+    "        features = set(X.columns) - {\"censored\"}\n",
+    "        dtrain = xgb.DMatrix(X[list(features)], enable_categorical=True)\n",
+    "        if set(np.unique(X[\"censored\"])) != {0, 1}:\n",
+    "            raise ValueError(\"censored column should be binary.\")\n",
+    "\n",
+    "        y_upper_bound = np.where(X[\"censored\"], +np.inf, y)\n",
+    "        y_lower_bound = np.array(y)\n",
+    "\n",
+    "        dtrain.set_float_info(\"label_lower_bound\", y_lower_bound)\n",
+    "        dtrain.set_float_info(\"label_upper_bound\", y_upper_bound)\n",
+    "\n",
+    "        if self.booster_params is None:\n",
+    "            booster_params = common_params\n",
+    "        else:\n",
+    "            booster_params = common_params | self.booster_params\n",
+    "\n",
+    "        self.bst_ = xgb.train(booster_params, dtrain, num_boost_round=self.n_estimators)\n",
+    "        return self\n",
+    "\n",
+    "    def predict(self, X):\n",
```

Comment:
Do we really need the `censored` column at prediction time? Couldn't we use a dummy value?

Reply:
We don't need it, actually it's just dropped, but then when calling `predict` we would need to make sure we don't pass it. This could give problems with `metalearners` as it assumes the structure of `X` is the same for `fit` and `predict`.

Reply:
I see, then it's fine! 👍